satpy/tests/reader_tests/test_ami_l1b.py

```diff
@@ -57,42 +57,59 @@ def close(self):
 class TestAMIL1bNetCDFBase(unittest.TestCase):
     """Common setup for NC_ABI_L1B tests."""
 
+    def __init__(self, *args):
+        """Initialize test data."""
+        super(TestAMIL1bNetCDFBase, self).__init__(*args)
+        self.counts = None
+        self.irtest = None
+
     @mock.patch("satpy.readers.ami_l1b.xr")
-    def setUp(self, xr_, counts=None):
+    def setUp(self, xr_):
         """Create a fake dataset using the given counts data."""
         from satpy.readers.ami_l1b import AMIL1bNetCDF
-
-        if counts is None:
+        if self.irtest:
+            dn_to_Radiance_Gain = -0.00108296517282724
+            dn_to_Radiance_Offset = 17.699987411499
+            bpp = 14
+        else:
+            dn_to_Radiance_Gain = -0.0144806550815701
+            dn_to_Radiance_Offset = 118.050903320312
+            bpp = 12
+
+        if self.counts is None:
             rad_data = (np.arange(10.).reshape((2, 5)) + 1.) * 50.
             rad_data = (rad_data + 1.) / 0.5
             rad_data = rad_data.astype(np.uint16)
-            counts = xr.DataArray(
-                da.from_array(rad_data, chunks="auto"),
-                dims=("y", "x"),
-                attrs={
-                    "channel_name": "VI006",
-                    "detector_side": 2,
-                    "number_of_total_pixels": 484000000,
-                    "number_of_error_pixels": 113892451,
-                    "max_pixel_value": 32768,
-                    "min_pixel_value": 6,
-                    "average_pixel_value": 8228.98770845248,
-                    "stddev_pixel_value": 13621.130386551,
-                    "number_of_total_bits_per_pixel": 16,
-                    "number_of_data_quality_flag_bits_per_pixel": 2,
-                    "number_of_valid_bits_per_pixel": 12,
-                    "data_quality_flag_meaning":
-                        "0:good_pixel, 1:conditionally_usable_pixel, 2:out_of_scan_area_pixel, 3:error_pixel",
-                    "ground_sample_distance_ew": 1.4e-05,
-                    "ground_sample_distance_ns": 1.4e-05,
-                }
-            )
+            if self.irtest:
+                # If testing IR clipping, set one pixel to negative radiance
+                rad_data[0, 0] = 16364
+                self.counts = xr.DataArray(
```

Comment:
I'm rewriting these tests, but I'm realizing that this block for `self.counts = ` is probably indented one level too deep so `self.counts` never gets set. I have no idea how your tests were working.

Reply:
Oh `self.counts` is set in the `setUp` :man_facepalming: nevermind then